src/services/quotes/quote-sources/sushiswap-quote-source.ts

```diff
@@ -0,0 +1,106 @@
+import { Chains, getChainByKey } from '@chains';
+import { QuoteParams, QuoteSourceMetadata, SourceQuoteResponse, SourceQuoteTransaction, BuildTxParams } from './types';
+import qs from 'qs';
+import { addQuoteSlippage, calculateAllowanceTarget, failed } from './utils';
+import { AlwaysValidConfigAndContextSource } from './base/always-valid-source';
+
+const SUSHISWAP_METADATA: QuoteSourceMetadata<SushiswapSupport> = {
+  name: 'Sushiswap',
+  supports: {
+    chains: [
+      Chains.ETHEREUM.chainId,
+      Chains.OPTIMISM.chainId,
+      Chains.CRONOS.chainId,
+      Chains.ROOTSTOCK.chainId,
+      Chains.BNB_CHAIN.chainId,
+      Chains.GNOSIS.chainId,
+      Chains.FUSE.chainId,
+      Chains.POLYGON.chainId,
+      Chains.SONIC.chainId,
+      Chains.FANTOM.chainId,
+      Chains.ZK_SYNC_ERA.chainId,
+      Chains.MOONBEAM.chainId,
+      Chains.MOONRIVER.chainId,
+      Chains.KAVA.chainId,
+      Chains.MANTLE.chainId,
+      Chains.BASE.chainId,
+      Chains.MODE.chainId,
+      Chains.ARBITRUM.chainId,
+      Chains.CELO.chainId,
+      Chains.LINEA.chainId,
+      Chains.BLAST.chainId,
+      Chains.SCROLL.chainId,
+    ],
+    swapAndTransfer: true,
+    buyOrders: false,
+  },
+  logoURI: '',
+};
+type SushiswapConfig = { apiKey: string };
+type SushiswapSupport = { buyOrders: false; swapAndTransfer: true };
+type SushiswapData = { tx: SourceQuoteTransaction };
+export class SushiswapQuoteSource extends AlwaysValidConfigAndContextSource<SushiswapSupport, SushiswapConfig, SushiswapData> {
+  getMetadata() {
+    return SUSHISWAP_METADATA;
+  }
+  async quote({
+    components: { fetchService },
+    request: {
+      chainId,
+      sellToken,
+      buyToken,
+      order,
+      accounts: { takeFrom, recipient },
+      config: { slippagePercentage, timeout },
+    },
+    config,
+  }: QuoteParams<SushiswapSupport, SushiswapConfig>): Promise<SourceQuoteResponse<SushiswapData>> {
+    const chain = getChainByKey(chainId);
+    if (!chain) throw new Error(`Chain with id ${chainId} not found`);
+    const queryParams = {
+      amount: order.sellAmount,
+      tokenIn: sellToken,
+      tokenOut: buyToken,
+      maxSlippage: slippagePercentage / 100,
+      sender: takeFrom,
+      recipient,
+
+      // Without this, the sender needs to have the balance to get the quote
+      simulate: false,
```

Comment:
I think you can do this
```suggestion
      simulate: !config.disableValidation,
```